### PR TITLE
test/kokoro: Migrate PSM Interop to Artifact Registry

### PR DESCRIPTION
**Pull request by sergiitk**
Migrate PSM Interop images from Container Registry (gcr.io) to Artifact Registry (pkg.dev).

RELEASE NOTES: n/a

### CODE REVIEW

**Comment by sergiitk** [comment]
- [ ] [grpc/go/master/branch/xds_k8s_lb](https://source.cloud.google.com/results/invocations/50cec78d-5e9e-4059-ac99-e35fd7b45b84)

**Comment by sergiitk** [comment]
- [ ] [grpc/go/master/branch/xds_k8s_lb](https://source.cloud.google.com/results/invocations/5edf93d4-908b-4761-a2e6-c43401d9e3f3)

**Comment by sergiitk** [comment]
The build works now. @arvindbr8 - please review.

File: test/kokoro/xds_k8s_lb.sh [hunk 1]
```diff
@@ -19,8 +19,9 @@ set -eo pipefail
 readonly GITHUB_REPOSITORY_NAME="grpc-go"
 readonly TEST_DRIVER_INSTALL_SCRIPT_URL="https://raw.githubusercontent.com/${TEST_DRIVER_REPO_OWNER:-grpc}/psm-interop/${TEST_DRIVER_BRANCH:-main}/.kokoro/psm_interop_kokoro_lib.sh"
 ## xDS test server/client Docker images
-readonly SERVER_IMAGE_NAME="gcr.io/grpc-testing/xds-interop/go-server"
-readonly CLIENT_IMAGE_NAME="gcr.io/grpc-testing/xds-interop/go-client"
+readonly DOCKER_REGISTRY="us-docker.pkg.dev"
```
Comment (arvindbr8) [inline test/kokoro/xds_k8s_lb.sh:22]: Doesnt seem like we are initializing `$DOCKER_REGISTRY` in the other .sh files

Comment (sergiitk) [reply test/kokoro/xds_k8s_lb.sh:22]: Yes, thanks for catching this! Fixed now.

**Pull request merged**